v7.11 Changelog (#6369)
mattermost/docs: source/install/self-managed-changelog.md

Latest Mattermost Releases:

- [Release v7.11 - Feature Release](#release-v7-11-feature-release)
- [Release v7.10 - Feature Release](#release-v7-10-feature-release)
- [Release v7.9 - Feature Release](#release-v7-9-feature-release)
- [Release v7.8 - Extended Support Release](#release-v7-8-extended-support-release)

## Release v7.11 - [Feature Release](https://docs.mattermost.com/upgrade/release-definitions.html#feature-release)

**Release day: May 16, 2023**

### Important Upgrade Notes

>  - In v7.11, the following repositories are merged into one: ``mattermost-server``, ``mattermost-webapp``, ``focalboard`` and ``mattermost-plugin-playbooks``. Developers should read the updated [Developer Guide](https://developers.mattermost.com/contribute/developer-setup/) for details. **Playbooks and Boards are no longer core parts of the product and cannot be disabled**.

Reviewer: Are Playbooks and Boards no longer **core** parts of the product? Or are they now core parts of the product (and therefore can't be disabled)? (based on updates made to important upgrade notes)

>  - Fixed an issue caused by a migration in the previous release. Query takes around 11ms on a PG 14 t3.medium RDS instance. Locks on the preferences table will only be acquired if there are rows to delete. But the time taken is negligible.

Reviewer: ```suggestion
 - Fixed an issue caused by a migration in the previous release. Query takes around 11ms on a PostgreSQL 14 DB t3.medium RDS instance. Locks on the preferences table will only be acquired if there are rows to delete, but the time taken is negligible.
```

**IMPORTANT:** If you upgrade from a release earlier than v7.10, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Improvements

#### User Interface (UI)
 - Enabled Work Templates in the Onboarding checklist and the `+` Menu.
 - Added a **Mattermost Marketplace** option to the bottom of the apps bar. The option is visible when the Marketplace is enabled, and the user has ``SYSCONSOLE_WRITE_PLUGINS`` permissions.
 - Added an **Add channels** button to the bottom of the left-hand sidebar to make the action more obvious for users who want to create or join channels.
 - Removed the Webapp Build Hash from **Main Menu > About Mattermost** since it is now identical to Server Build Hash.
 - Replaced the ``compass-components`` icon component with ``compass-icons``.
 - Added “hours ahead” timezone details to the user profile popover.

#### Administration

>  - Self-hosted admins can now choose a separate shipping address.

Reviewer: ```suggestion
 - Self-hosted admins can now define a separate shipping address.
```

Reviewer: Where can admins set a shipping address?

Reviewer 2: I added a suggestion below so that it's clear that is during in-product license purchase

>  - Self-hosted admins can now define a separate shipping address.

Reviewer 2: ```suggestion
 - Self-hosted admins can now define a separate shipping address during in-product license purchase.
```

 - Added updates to the trial request forms to allow for a more tailored trial experience.

>  - Added a form page to capture company name data and the invite members section to the preparing workspace screen for self-hosted installations.

Reviewer: ```suggestion
 - Added a form page to capture company name data, and added the invite members section to the Preparing Workspace screen for self-hosted installations.
```

>  - Added a form page to capture company name data, and added the invite members section to the Preparing Workspace screen for self-hosted installations.

Reviewer 2: ```suggestion
 - First admins will now have an onboarding experience that includes first team creation based on company name and invite members link steps. 
```

 - Adds the ability to expand seats in-product for self-hosted servers.

>  - Added a new section in the **System Console** for products.

Reviewer 2: We should probably explain this a bit better because at first I though products was like SKUs but I believe this is referring to Boards and Playbooks?

 - The Go MySQL driver has changed the ``maxAllowedPacket`` size from 4MiB to 64MiB. This is to make it consistent with the change in the server side default value from MySQL 5.7 to MySQL 8.0. If your ``max_allowed_packet`` setting is not 64MiB, then please update the MySQL config DSN with an additional param of ``maxAllowedPacket`` to match with the server side value. Alternatively, a value of 0 can be set to to automatically fetch the server side value, on every new connection, which has a performance overhead.
 - Removed ``ExperimentalSettings.PatchPluginsReactDOM``. If this setting was previously enabled, confirm that:

    1. All Mattermost-supported plugins are updated to the latest versions.
    2. Any other plugins have been updated to support React 17. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for v7.7 for more information.

 - Removed deprecated ``PermissionUseSlashCommands``.
 - Removed deprecated ``model.CommandArgs.Session``.
 - Pass a ``context.Context`` to Client4 methods.

>  - For servers wanting to allow websockets to connect from other origins, please set the ``ServiceSettings.AllowCorsFrom`` config setting.

Reviewer: Config setting docs link: https://docs.mattermost.com/configure/integrations-configuration-settings.html#enable-cross-origin-requests-from

- [Release v7.9 - Feature Release](#release-v7-9-feature-release)
- [Release v7.8 - Extended Support Release](#release-v7-8-extended-support-release)

## Release v8.0 - [Major Release](https://docs.mattermost.com/upgrade/release-definitions.html#major-release)

**Release Day: July 14, 2023**

### Important Upgrade Notes

 - Insights has been disabled for all new instances and for existing servers that upgrade to v8.0. See more details in [this forum post](https://forum.mattermost.com/t/proposal-to-revise-our-insights-feature-due-to-known-performance-issues/16212) on why Insights has been disabled.
 - The Focalboard plugin is now disabled by default and can be enabled in the **System Console > Plugin settings**.
 - The Channel Export and Apps plugins are now disabled by default.
 - Apps Bar is now enabled by default for on-prem servers. ``ExperimentalSettings.EnableAppBar`` was also renamed to ``ExperimentalSettings.DisableAppBar``. See more details at:
   - https://docs.mattermost.com/configure/experimental-configuration-settings.html#disable-app-bar 
   - https://forum.mattermost.com/t/channel-header-plugin-changes/13551

>  - The Go module has been upgraded to v8.0. All packages are now under the new path ``github.com/mattermost-server/server/v8``. Clients will have to change their module path to the new structure. Very minimal change. There are no API level breaking changes that would require code changes.

Reviewer 3: This can be removed. https://github.com/mattermost/docs/pull/6369/files#diff-eba248b2c1a66750343693c8d84e6909fc3c26f299880afcc7261837a0465323R28-R29 covers the go module changes.
```suggestion
```

 - The file info stats query is now optimized by denormalizing the ``channelID`` column into the table itself. This will speed up the query to get the file count for a channel when selecting the right-hand pane. Migration times:

   - On a MySQL 8.0.31 DB with 1405 rows in FileInfo and 11M posts, it took around 0.3s
   - On a PostgreSQL 12.14 DB with 1731 rows in FileInfo and 11M posts, it took around 0.27s

**IMPORTANT:** If you upgrade from a release earlier than v7.10, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

#### Persistent Notifications
 - Persistent notifications allow users to notify recipients repeatedly until action is taken on an urgent message. Check out our documentation for more details.

#### Apps Bar
 - The apps bar is now enabled by default for on-prem servers. ``ExperimentalSettings.EnableAppBar`` was also renamed to ``ExperimentalSettings.DisableAppBar``. See more details at:
   - https://docs.mattermost.com/configure/experimental-configuration-settings.html#disable-app-bar 
   - https://forum.mattermost.com/t/channel-header-plugin-changes/13551

#### Calls

>  - 

Reviewer 4: @streamer @cwarnermm , I propose:
```suggestion
 - Calls v0.17.0 introduces a new ringing feature: Calls in DM and GM channels will ring and pop up a visual notification for the incoming call. Check out the Calls v0.17.0 release notes and Calls documentation for more details.
```

>  - 

Reviewer: Let's spell out DM and GM as direct messages and group messages for clarity.

Is this a beta feature? Should it be labeled as such in the documentation? Also, is this feature enabled by default for new and upgraded deployments?

>  - 

Author: cc @cpoile for the above questions.

>  - 

Reviewer 5: @cwarnermm Yes the feature will be labeled as beta and as such it will be off by default. Needs to be explicitly enabled through the admin console.

>  - 

Reviewer 4: @cwarnermm Should we add this to the release notes then? (It looks like it wasn't)

>  - 

Reviewer 4: @cwarnermm nvm, I see the change now. Thank you!

Latest Mattermost Releases:

- [Release v8.0 - Major Release](#release-v8-0-major-release)
- [Release v7.11 - Feature Release](#release-v7-11-feature-release)
- [Release v7.10 - Feature Release](#release-v7-10-feature-release)
- [Release v7.9 - Feature Release](#release-v7-9-feature-release)
- [Release v7.8 - Extended Support Release](#release-v7-8-extended-support-release)

## Release v8.0 - [Major Release](https://docs.mattermost.com/upgrade/release-definitions.html#major-release)

**Release Day: July 14, 2023**

### Important Upgrade Notes

 - Insights has been disabled for all new instances and for existing servers that upgrade to v8.0. See more details in [this forum post](https://forum.mattermost.com/t/proposal-to-revise-our-insights-feature-due-to-known-performance-issues/16212) on why Insights has been disabled.

>  - Boards is now disabled by default and can be enabled in the **System Console > Plugin settings**.

Reviewer 6: The Focalboard plugin is now disabled by default and can be enabled in the **System Console > Plugin settings**

 - Fixed an issue caused by a migration in the previous release. Query takes around 11ms on a PostgreSQL 14 DB t3.medium RDS instance. Locks on the preferences table will only be acquired if there are rows to delete, but the time taken is negligible.
 - Fixed an issue where a user would still see threads in the threads view of channels they have left. Migration execution time in MySQL: Query OK, 2766769 rows affected (4 min 47.57 sec). Migration execution time in PostgreSQL: Execution time: 58.11 sec, DELETE 2766690.
 - The file info stats query is now optimized by denormalizing the ``channelID`` column into the table itself. This will speed up the query to get the file count for a channel when selecting the right-hand pane. Migration times:

   - On a MySQL 8.0.31 DB with 1405 rows in FileInfo and 11M posts, it took around 0.3s
   - On a PostgreSQL 12.14 DB with 1731 rows in FileInfo and 11M posts, it took around 0.27s

**IMPORTANT:** If you upgrade from a release earlier than v7.10, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

#### Persistent Notifications

>  - Added a persistent notification option when sending urgent priority posts.

Reviewer 6: Persistent notifications allow users to notify recipients repeatedly until action is taken on an urgent message. Check out our documentation for more details. [Link]

#### Apps Bar
 - The apps bar is now enabled by default for on-prem servers. ``ExperimentalSettings.EnableAppBar`` was also renamed to ``ExperimentalSettings.DisableAppBar``. See more details at:
   - https://docs.mattermost.com/configure/experimental-configuration-settings.html#disable-app-bar 
   - https://forum.mattermost.com/t/channel-header-plugin-changes/13551

#### Calls
 - 

#### Playbooks
 - 

> #### Boards

Reviewer 6: Focalboard

>  - Prepackaged Boards version 7.11.0.

Reviewer 6: Boards > Focalboard plugin

 - Added “hours ahead” timezone details to the user profile popover.
 - Added an experimental feature to disable re-fetching of channel and channel members on browser focus.
 - Bot users are now hidden in the user selector in apps forms.
 - Removed the fetching of archived channels on page load.
 - The **Channel Type** dropdown within the **Browse Channels** modal can now be focused.
 - Removed in-app help pages that were no longer accessible.
 - Removed system join/leave messages from thread replies and post them instead in the main channel.
 - Added an experimental setting to make the channel autocomplete only appear after typing two characters instead of immediately after the tilde (~).
 - Users with default profile pictures will now regenerate a new picture when their username is changed.
 - Implemented URL auto generation on channel creation for when there's no URL safe characters on its name.
 - Added a new option to auto-follow all threads in the channel **Notification Preference** settings.
 - ``CTRL/CMD + K`` shortcut can now be used to insert link formatting when text is selected.
 - ``pas`` and ``pascal`` code blocks are now higlighted.

>  - Language setting in Boards was removed. The main language setting under **Settings -> Display Settings** now covers Boards. Boards previously supported Catalan, Greek, Indonesian, and Occitan, but since these 4 languages were only partially translated (<40%; Boards-only), they have been removed until more areas of Mattermost are translated into these languages.

Reviewer 6: Propose remove from changelog since it's related to the focalboard plugin itself.

I propose these types of plugin changes go in the v7.11 focalboard changelog rather than the server changelog

Author: @sbishel Is this and the below notes from Eric already in the Focalboard changelog?

    - ``save-plan``: The plan for the migration will be saved into the file store so that it can be used for reviewing the plan or to be used for downgrading.
    - ``dry-run``: Does not apply the migrations, but it validates how the migration would run with the given conditions.

 - A new [database subcommand](https://docs.mattermost.com/manage/command-line-tools.html#mattermost-db-downgrade) "downgrade" was added to be able to rollback database migrations. The command either requires an update plan to rollback, or comma separated version numbers.
 - Removed ``/api/v4/users/stats`` network request from ``InviteMembersButton``.
 - Self-hosted admins can now define a separate shipping address during in-product license purchase.
 - Added updates to the trial request forms to allow for a more tailored trial experience.
 - First admins will now have an onboarding experience that includes first team creation based on company name and invite members link steps. 
 - Adds the ability to expand seats in-product for self-hosted servers.

>  - Added a new section in the **System Console > Products** for Boards.

Reviewer 6: Shouldn't this setting be in the plugin settings @sbishel rather than in a new system console section? If you agree perhaps we should add a ticket for v9.0

 - Added the ability to search a partial first name, last name, nickname, or username on the **System Console > Users** page.
 - **Contact Support** now redirects users to Zendesk and pre-fills known information.
 - Added a mechanism for public routes on products and used it to support publicly shared Board links.
 - The database section in the **System Console** now has an additional read-only section which shows the active search backend in use. This can be helpful to confirm which search engine is currently active when there are multiple configured.
 - Updated Docker Base Image from Debian to Ubuntu 22.04 LTS.
 - Type-generated settings will now be generated (only for future generations) with a URL-safe version of base64 encoding.
 - Mattermost is now resilient against database replica outages and will dynamically choose a replica if it's alive. Also a config parameter ``ReplicaMonitorIntervalSeconds`` was added and the default value is 5. This controls how frequently unhealthy replicas will be monitored for liveness check.

>  - Added support for attachments when importing/exporting Boards.

Reviewer 6: move to focalboard changelog for focalboard v7.11

 - Updated Docker Base Image from Debian to Ubuntu 22.04 LTS.
 
#### Performance
 - Improved the performance of webapp related to timezone calculations.
 - Improved performance of code used for post list screen reader support.

### API Changes
 - An underscore is now used in the timeline API (``event-id`` -> ``event_id``) for consistency with other API arguments.

### Bug Fixes
 - Fixed a scrolling issue in the purchase modals.

>  - Fixed an issue where the **Delete Category Dialog** message was not visible in Boards.

Reviewer 6: move to focalboard changelog for focalboard v7.11

 - Fixed an issue where the experimental Shared Channels feature failed to synchronize if a previously removed table column was still present.

>  - Fixed an innocuous panic in Boards Rest API when requesting files and an error other than ``not found`` is encountered.

Reviewer 6: move to focalboard changelog for focalboard v7.11

    1. All Mattermost-supported plugins are updated to the latest versions.
    2. Any other plugins have been updated to support React 17. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for v7.7 for more information.

 - Removed deprecated ``PermissionUseSlashCommands``.
 - Removed deprecated ``model.CommandArgs.Session``.
 - Pass a ``context.Context`` to Client4 methods.
 - For servers wanting to allow websockets to connect from other origins, please set the ``ServiceSettings.AllowCorsFrom`` [config setting](https://docs.mattermost.com/configure/integrations-configuration-settings.html#enable-cross-origin-requests-from).
 - In v8.0, the following repositories are merged into one: ``mattermost-server``, ``mattermost-webapp`` and ``mmctl``. Developers should read the updated [Developer Guide](https://developers.mattermost.com/contribute/developer-setup/) for details.
 - Fixed an issue caused by a migration in the previous release. Query takes around 11ms on a PostgreSQL 14 DB t3.medium RDS instance. Locks on the preferences table will only be acquired if there are rows to delete, but the time taken is negligible.

>  - Fixed an issue where a user would still see threads in the threads view of channels they have left. Migration execution time in MySQL: Query OK, 2766769 rows affected (4 min 47.57 sec). Migration execution time in PostgreSQL: Execution time: 58.11 sec, DELETE 2766690.

Reviewer: Suggestion: start listing PostgreSQL details ahead of MySQL details across all product documentation.

**IMPORTANT:** If you upgrade from a release earlier than v7.10, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Highlights

#### Private cloud LLMs, Azure AI and OpenAI integrations
 - Mattermost provides an OpenOps framework to integrate with private cloud LLMs, Azure AI and OpenAI models to embed generative AI assistance in collaborative workflows and automation. [Learn more about OpenOps here](https://github.com/mattermost/openops).

#### Mattermost for Microsoft Teams
 - We’re extending our integration with the Microsoft 365 platform with a new embedded experience directly inside Microsoft Teams, as well as our updated MS Teams Connector.

#### Mattermost for Atlassian Suite
 - Uplevel your workflows within Mattermost using your Atlassian toolset. [Learn more about Mattermost for Atlassian Suite here](https://mattermost.com/atlassian/).

#### Performance and efficiency with PostgreSQL 
 - To simplify management and scalability challenges, Mattermost 8.0 recommends deploying PostgreSQL over MySQL.

#### New End User Training

>  - We’re introducing [9 new training modules](https://academy.mattermost.com/p/mattermost-end-user-onboarding) dedicated to educating users on the key components of the Mattermost platform and an additional [10 new use case modules](https://academy.mattermost.com/courses/category/use-case-training) tackling technical scenarios within DevOps, Security Ops, and Incident Management.

Reviewer: ```suggestion
 - We’re introducing [10 new training modules](https://academy.mattermost.com/p/mattermost-end-user-onboarding) dedicated to educating users on the key components of the Mattermost platform and an additional [10 new use case modules](https://academy.mattermost.com/courses/category/use-case-training) tackling technical scenarios within DevOps, Security Ops, and Incident Management.
```

### Improvements

#### User Interface (UI)
 - Persistent notifications allow users to notify recipients repeatedly until action is taken on an urgent message. Check out [our documentation](https://docs.mattermost.com/channels/message-priority.html#send-persistent-notifications) for more details.
 - The apps bar is now enabled by default for on-prem servers. ``ExperimentalSettings.EnableAppBar`` was also renamed to ``ExperimentalSettings.DisableAppBar``. See more details at:
   - https://docs.mattermost.com/configure/experimental-configuration-settings.html#disable-app-bar 
   - https://forum.mattermost.com/t/channel-header-plugin-changes/13551
 - Added a **Mattermost Marketplace** option to the bottom of the apps bar. The option is visible when the Marketplace is enabled, and the user has ``SYSCONSOLE_WRITE_PLUGINS`` permissions.

>  - Calls v0.17.0 introduces a new ringing feature (Beta): Calls in Direct and Group Message channels will ring and pop up a visual notification for the incoming call. Check out the Calls v0.17.0 release notes and Calls documentation for more details.

Reviewer: Calls docs: https://docs.mattermost.com/channels/make-calls.html

 - Added an **Add channels** button to the bottom of the left-hand sidebar to make the action more obvious for users who want to create or join channels.
 - Removed the Webapp Build Hash from **Main Menu > About Mattermost** since it is now identical to Server Build Hash.
 - Replaced the ``compass-components`` icon component with ``compass-icons``.
 - Added “hours ahead” timezone details to the user profile popover.
 - Added an experimental feature to disable re-fetching of channel and channel members on browser focus.
 - Bot users are now hidden in the user selector in apps forms.
 - Removed the fetching of archived channels on page load.
 - The **Channel Type** dropdown within the **Browse Channels** modal can now be focused.
 - Removed in-app help pages that were no longer accessible.
 - Removed system join/leave messages from thread replies and post them instead in the main channel.

>  - Added an experimental setting to make the channel autocomplete only appear after typing two characters instead of immediately after the tilde (~).

Reviewer: Delay channel autocomplete docs: https://docs.mattermost.com/configure/experimental-configuration-settings.html#delay-channel-autocomplete

 - Fixed a panic where a JSON null value was passed as a channel update.
 - Fixed an issue where the draft counter badge remained in cases where a deleted parent post was removed.
 - Fixed an issue where posts were not fully sanitized for audit output when a link preview was included.
 - Fixed an issue where the footer with **Save/Cancel** buttons did not get anchored properly in the System Console.
 - Fixed an issue where the undo history was erased when links, tables, or code was pasted into the textbox.
 - Fixed an issue where Elasticsearch didn't properly start on startup when enabled. Also added a missing ``IsEnabled`` method to Elasticsearch.
 - Fixed an issue where text couldn't be copied from the post textbox.
 
### config.json
Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.

#### Changes to Team Edition and Enterprise Edition:
 - Removed ``EnableInactivityEmail`` config setting.
 - Added a new config setting section ``ProductSettings``.
 - Under ``ServiceSettings`` in ``config.json``:

>    - Added new confic settings ``AllowPersistentNotifications``, ``AllowPersistentNotificationsForGuests``, ``PersistentNotificationIntervalMinutes``, ``PersistentNotificationMaxCount``, ``PersistentNotificationMaxRecipients``, to add a persistent notification option when sending urgent priority posts.

Reviewer: ```suggestion
   - Added new configuration settings ``AllowPersistentNotifications``, ``AllowPersistentNotificationsForGuests``, ``PersistentNotificationIntervalMinutes``, ``PersistentNotificationMaxCount``, ``PersistentNotificationMaxRecipients``, to add a persistent notification option when sending urgent priority posts.
```

Reviewer: Note: ``AllowPersistentNotificationsForGuests`` wasn't implemented and should be removed. CC @esethna

 - Added an experimental feature to disable re-fetching of channel and channel members on browser focus.
 - Bot users are now hidden in the user selector in apps forms.
 - Removed the fetching of archived channels on page load.
 - The **Channel Type** dropdown within the **Browse Channels** modal can now be focused.
 - Removed in-app help pages that were no longer accessible.
 - Removed system join/leave messages from thread replies and post them instead in the main channel.
 - Added an experimental setting to make the channel autocomplete only appear after typing two characters instead of immediately after the tilde (~).
 - Users with default profile pictures will now regenerate a new picture when their username is changed.
 - Implemented URL auto generation on channel creation for when there's no URL safe characters on its name.
 - Added a new option to auto-follow all threads in the channel **Notification Preference** settings.
 - ``CTRL/CMD + K`` shortcut can now be used to insert link formatting when text is selected.
 - ``pas`` and ``pascal`` code blocks are now higlighted.
 - Removed websocket state effects for the collapse/expand state of categories.

>  - Pre-packaged Calls version 0.16.1.

Reviewer 5: Pre-packaged Calls version in 8.0 is `0.17.0`
